RubberduckTests/CodeExplorer/CodeExplorerComponentViewModelTests.cs

```diff
@@ -0,0 +1,332 @@
+            var componentDeclaration = declarations
+                .First(declaration => declaration.DeclarationType.HasFlag(DeclarationType.Module) &&
+                    declaration.IdentifierName.Equals(CodeExplorerTestSetup.TestModuleName));
+
+            var component = new CodeExplorerComponentViewModel(null, componentDeclaration, declarations, null);
+
+            Assert.AreEqual(CodeExplorerItemComparer.ComponentType.GetType(), component.SortComparer.GetType());
+        }
+
+        [Test]
+        [Category("Code Explorer")]
+        [TestCase(CodeExplorerTestSetup.TestDocumentName)]
+        [TestCase(CodeExplorerTestSetup.TestModuleName)]
+        [TestCase(CodeExplorerTestSetup.TestClassName)]
+        [TestCase(CodeExplorerTestSetup.TestUserFormName)]
+        public void FilteredIsFalseForSubsetsOfName(string name)
+        {
+            var componentDeclaration = CodeExplorerTestSetup.TestProjectOneDeclarations
+                .First(declaration => declaration.DeclarationType.HasFlag(DeclarationType.Module) && declaration.IdentifierName.Equals(name));
+
+            var component = new CodeExplorerComponentViewModel(null, componentDeclaration, new List<Declaration> { componentDeclaration }, null);
+
+            for (var characters = 1; characters <= name.Length; characters++)
+            {
+                component.Filter = name.Substring(0, characters);
+                Assert.IsFalse(component.Filtered);
+            }
+
+            for (var position = name.Length - 2; position > 0; position--)
+            {
+                component.Filter = name.Substring(position);
+                Assert.IsFalse(component.Filtered);
+            }
+        }
+
+        [Test]
+        [Category("Code Explorer")]
+        [TestCase(CodeExplorerTestSetup.TestDocumentName)]
+        [TestCase(CodeExplorerTestSetup.TestModuleName)]
+        [TestCase(CodeExplorerTestSetup.TestClassName)]
+        [TestCase(CodeExplorerTestSetup.TestUserFormName)]
+        public void FilteredIsTrueForCharactersNotInName(string name)
+        {
+            const string testCharacters = "abcdefghijklmnopqrstuwxyz";
+
+            var componentDeclaration = CodeExplorerTestSetup.TestProjectOneDeclarations
+                .First(declaration => declaration.DeclarationType.HasFlag(DeclarationType.Module) && declaration.IdentifierName.Equals(name));
+
+            var component = new CodeExplorerComponentViewModel(null, componentDeclaration, new List<Declaration> { componentDeclaration }, null);
+
+            var nonMatching = testCharacters.ToCharArray().Except(name.ToLowerInvariant().ToCharArray());
+
+            foreach (var character in nonMatching.Select(letter => letter.ToString()))
+            {
+                component.Filter = character;
+                Assert.IsTrue(component.Filtered);
+            }
```

Comment:
Ought there be an `Assert` for the matching? (I'm understanding the test to contain only one matching result for each iteration).